Service/src/routes/league.py

```diff
@@ -1,38 +1,43 @@
 from flask import Blueprint, request, Response, g, jsonify, make_response
 import simplejson as json
-import time
+from datetime import datetime
 
+from auth import auth
 import routes.portfolio as portfolio
+from request_validator import validator
+from request_validator.schemas import league_post_schema
 from util.utility import Utility
 from util.error_map import errors
 
 league_api = Blueprint('league_api', __name__)
 
 DATE_FORMAT = "%m-%d-%Y"
+DAYS_IN_YEAR = 365
 
-
-@league_api.route('/api/league', methods=['POST'])
+@league_api.route('/api/v1.0/leagues', methods=['POST'])
+@auth.login_required
+@validator.validate_body(league_post_schema.fields)
 def post_league():
    body = request.get_json()
-   try:
-      result = LeagueSchema().load(body)
-   except ValidationError as err:
-      return make_response(json.dumps(err.messages), 400)
-
+   startDate = datetime.strptime(body['start'], DATE_FORMAT)
+   endDate = datetime.strptime(body['end'], DATE_FORMAT)
+   if startDate > endDate:
+      return make_response(jsonify(EndBeforeStart=errors['endBeforeStart']), 400)
+   
+   leagueDuration = (endDate - startDate).days
+   print(str(leagueDuration))
+   if leagueDuration > DAYS_IN_YEAR:
```

Comment:
Why do we need to limit it to a year? Also, can we have leagues that don't have an end date (for those personal portfolios for users to just play with)?

Reply:
I just wanted to set a limit for the sake of setting a limit. I'm gonna keep this here for now and if we decide in a meeting we don't want any limit I'll remove it.